framework/db/Query.php

```diff
@@ -466,18 +466,29 @@ public function getTablesUsedInFrom()
     {
         if (empty($this->from)) {
             return [];
-        } elseif (is_array($this->from)) {
+        }
+
+        if (is_array($this->from)) {
             $tableNames = $this->from;
         } elseif (is_string($this->from)) {
             $tableNames = preg_split('/\s*,\s*/', trim($this->from), -1, PREG_SPLIT_NO_EMPTY);
+        } elseif ($this->from instanceof Expression) {
```

Comment:
Should it be replaced with `ExpressionInterface` after merging that PR?

Reply:
Yes.